Tests/SourceKitLSPTests/InlayHintsTests.swift

```diff
@@ -0,0 +1,159 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import LanguageServerProtocol
+import LSPTestSupport
+import SKTestSupport
+import SourceKitLSP
+import XCTest
+
+final class InlayHintsTests: XCTestCase {
+  /// Connection and lifetime management for the service.
+  var connection: TestSourceKitServer! = nil
+
+  /// The primary interface to make requests to the SourceKitServer.
+  var sk: TestClient! = nil
+
+  override func tearDown() {
+    sk = nil
+    connection = nil
+  }
+
+  override func setUp() {
+    connection = TestSourceKitServer()
+    sk = connection.client
+    _ = try! sk.sendSync(InitializeRequest(
+      processId: nil,
+      rootPath: nil,
+      rootURI: nil,
+      initializationOptions: nil,
+      capabilities: ClientCapabilities(workspace: nil, textDocument: nil),
+      trace: .off,
+      workspaceFolders: nil
+    ))
+  }
+
+  func performInlayHintsRequest(text: String, range: Range<Position>? = nil) -> [InlayHint] {
+    let url = URL(fileURLWithPath: "/\(#function)/a.swift")
+    
+    sk.send(DidOpenTextDocumentNotification(textDocument: TextDocumentItem(
+      uri: DocumentURI(url),
+      language: .swift,
+      version: 17,
+      text: text
+    )))
+
+    let request = InlayHintsRequest(textDocument: TextDocumentIdentifier(url), range: range)
+    return try! sk.sendSync(request)
+  }
+
+  func testEmpty() {
+    let text = ""
+    let hints = performInlayHintsRequest(text: text)
+    XCTAssertEqual(hints, [])
+  }
+
+  func testBindings() {
+    let text = """
+    let x = 4
+    var y = "test" + "123"
+    """
+    let hints = performInlayHintsRequest(text: text)
+    XCTAssertEqual(hints, [
```

Comment:
Could you add a `FIXME` to these tests that the type hint should be displayed after the variable name if possible instead of after the expression?